src/Device/DeviceSelector/DeviceList/Device.jsx

```diff
@@ -0,0 +1,102 @@
+ *
+ * All rights reserved.
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useState } from 'react';
+import {
+    arrayOf, bool, func, shape, string,
+} from 'prop-types';
+import { serialports } from '../../deviceInfo/deviceInfo';
+import PseudoButton from '../../../PseudoButton/PseudoButton';
+import deviceShape from '../deviceShape';
+import BasicDeviceInfo from '../BasicDeviceInfo';
+
+import './device.scss';
+
+const Serialports = ({ ports }) => (
+    <ul className="ports">
+        {ports.map(port => <li key={port.path}>{port.path}</li>)}
+    </ul>
+);
+Serialports.propTypes = {
+    ports: arrayOf(
+        shape({
+            path: string.isRequired,
+        }).isRequired,
+    ).isRequired,
+};
+
+const MoreDeviceInfo = ({ device }) => (
+    <Serialports ports={serialports(device)} />
+);
+MoreDeviceInfo.propTypes = {
+    device: deviceShape.isRequired,
+};
+
+const additionalClassName = (moreVisible, isSelected) => {
+    if (moreVisible) return 'more-infos-visible';
+    if (isSelected) return 'selected-device';
+    return '';
+};
+
+const Device = ({ device, isSelected, doSelectDevice }) => {
```

Comment:
If I read this right currently you can expand the device to show the list of its serialports. The idea is that user can select serialports individually. Will such functionality be part of this PR? I think optionally extending the result with an index or serialport.path would be perfect.

Reply:
I don't understand this aspect about serial ports yet and so was too ignorant to implement it. Probably would help me, if I can see it in action: In which app can users select serial ports individually?

Reply:
Link monitor and Trace collector. All 91 gadgets have 2 or 3 serial ports and since it's just a guess which one is the correct, the user should be able to select if auto-selection does not work. See "Auto device/port filter" functionality of these apps.

Reply:
Ah, yes, let us care about them later and also discuss a good UI for that case with Ketil when we update one of these apps.

Reply:
You are right about that this is not yet needed, however we have an upcoming 91 app that will most likely require the same functionality. Agreed, not needed in this PR though.